test/channel.js

```diff
@@ -75,13 +75,11 @@ var DELIVER_FIELDS = {
   replyText: 'derp',
 };
 
-function open(ch) {
-  return Promise.try(function() {
-    ch.allocate();
-    return Promise.fromCallback(function(cb) {
-      ch._rpc(defs.ChannelOpen, {outOfBand: ''}, defs.ChannelOpenOk, cb);
-    });
-  });
+async function open(ch) {
```

Comment:
why make it async?

Reply:
My understanding of `Promise.try` is it makes all exceptions, synchronous or asynchronous, to get thrown in `.catch`. Making the function async does the same thing by wrapping the whole code in a promise.

Reply:
I'm confused by how the open was originally being used. There are some cases when `then` is never called, so the ChannelOpen rpc may execute after the code has continued.

I'm also not sure why Promise.try was required in the first place. ch.allocate can throw a synchronous error, but never does when I run the test suite. If we're uncomfortable with making the function async, then both the following appear to work...

```js
// Decide that Promise.try wasn't necessary to start with
function open(ch) {
  ch.allocate();
  return promisify(function(cb) {
    ch._rpc(defs.ChannelOpen, {outOfBand: ''}, defs.ChannelOpenOk, cb);
  })();
}
```

```js
// Emulate Promise.try without async
function open(ch) {
  return new Promise((resolve, reject) => {
    try {
      ch.allocate();
      ch._rpc(defs.ChannelOpen, {outOfBand: ''}, defs.ChannelOpenOk, (err, channel) => {
        if (err) return reject(err);
        resolve(channel);
      });
    } catch (err) {
      reject(err);
    }
  })
}
```

Reply:
async is redundant if we are returning Promise already, which I believe we are

Reply:
In the original code if ch.allocate throws an error while still synchronous, it will be thrown rather than rejected. Promise.try prevents this by wrapping the supplied function in a try/catch and rejecting anyway. That's the only reason for using Promise.try that I can see, but it doesn't seem to be necessary in this case

Reply:
I've removed the `async` since it seems unnecessary.